service/history/workflow/mutable_state_impl.go

```diff
@@ -218,6 +218,11 @@ type (
 		activityInfosUserDataUpdated map[int64]struct{}
 		timerInfosUserDataUpdated    map[string]struct{}
 
+		// in memory fields to track potential reapply events that needs to be reapplied during workflow update
+		// should only be used in the state based replication as state based replication does not have
+		// event inside history builder
+		reapplyEventsCandidate []*historypb.HistoryEvent
```

Comment:
nit: comment that this is only for x-run reapply (from zombie wf to current wf).

Reply:
updated.